src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AwsSdk/AwsSdkPipelineWrapper.cs

```diff
@@ -106,24 +103,24 @@ public AfterWrappedMethodDelegate BeforeWrappedMethod(InstrumentedMethodCall ins
                         }
                         else
                         {
-                            SqsHelper.InsertDistributedTraceHeaders(transaction, message);
+                            SqsHelper.InsertDistributedTraceHeaders(transaction, message, dtHeaders);
                         }
                     }
                 }
             }
 
-            // modify the request to ask for DT headers in the response message attributes
+            // modify the request to ask for DT headers in the response message attributes.
+            // we can't know which header(s) the message might contain, so we have to ask for all of them
             if (action == MessageBrokerAction.Consume)
             {
                 if (request.MessageAttributeNames == null)
                     request.MessageAttributeNames = new List<string>();
 
-                request.MessageAttributeNames.Add(NEWRELIC_TRACE_HEADER);
-                request.MessageAttributeNames.Add(W3C_TRACESTATE_HEADER);
-                request.MessageAttributeNames.Add(W3C_TRACEPARENT_HEADER);
+                request.MessageAttributeNames.Add(Constants.DistributedTracePayloadKeyAllLower);
```

Comment:
Should we just add all of the headers returned by the GetConfiguredDTHeaders call here? We are already calling it for the produce action.